components/NotificationsCard.tsx

```diff
@@ -2,326 +2,69 @@ import useRealm from '../hooks/useRealm'
+    data?.targetGroup?.smsTargets?.[0]?.phoneNumber ?? ''
+  )
+  const [telegram, setTelegram] = useState<string>(
+    data?.targetGroup?.telegramTargets?.[0]?.telegramId ?? ''
+  )
 
   const handleError = (errors: { message: string }[]) => {
     const message = errors.length > 0 ? errors[0]?.message : 'Unknown error'
     setErrorMessage(message)
     setLoading(false)
   }
 
-  const { jwtRef, setJwt } = useNotifiJwt()
-  const logInFromDao = useLoginFromDao()
   const handleSave = async function () {
     setLoading(true)
     // user is not authenticated
-    if (!jwtRef.current && wallet && wallet.publicKey) {
-      console.log(wallet)
-      console.log(jwtRef.current)
-      const ticks = Math.round(Date.now() / 1000)
-      const p = await (wallet as any).signMessage(
-        new TextEncoder().encode(
-          `${wallet?.publicKey}` + realm?.pubkey + ticks.toString()
-        ),
-        'utf8'
-      )
-
+    if (!isAuthenticated() && wallet && wallet.publicKey) {
       try {
-        const resp = logInFromDao({
+        await logIn({
           walletPublicKey: wallet.publicKey.toString(),
           daoAddress: realm?.pubkey.toBase58() || '',
-          timestamp: ticks,
-          signature: bufferToBase64(p),
         })
-
-        console.log(resp)
-
-        getExistingTargetGroup()
-        getFilter()
-        getSourceGroup()
       } catch (e) {
         handleError([e])
       }
       setLoading(false)
     }
 
-    if (connected && jwtRef.current) {
+    if (connected && isAuthenticated()) {
       console.log('Sending')
       console.log(email)
-      console.log(storedEmail)
-      console.log(storedEmailId)
-      let emailId = storedEmailId
-      let smsId = storedSmsId
-      let telegramId = storedTelegramId
       try {
-        if (email != storedEmail) {
-          console.log('creating email')
-          emailId = await createEmailTarget()
-          setStoredEmailId(emailId)
-        }
-
-        if (phone != storedSms) {
-          smsId = await createSmsTarget()
-          setStoredSmsId(smsId)
-        }
-
-        if (telegram != storedTelegram) {
-          telegramId = await createTelegramTarget()
-          setStoredTelegramId(telegramId)
-        }
-
-        if (targetGroup) {
-          // Update
-          console.log('updating target group')
-          await updateTargetGroup(emailId, smsId, telegramId)
-        } else {
-          // New
-          console.log('creating new target group')
-          const tgId = await createNewTargetGroup(emailId, smsId, telegramId)
-          setTargetGroup(tgId)
-          console.log('creating new alert')
-          await createAlert(tgId)
-        }
+        await updateAlert({
+          name: `${realm?.account.name} notifications`,
+          emailAddress: email === '' ? null : email,
+          phoneNumber: phone === '' ? null : phone,
+          telegramId: telegram === '' ? null : telegram,
+        })
       } catch (e) {
-        setJwt(null)
```

Comment:
Say if there's a stale JWT being used which causes an access denied, what's the expected flow?

Reply:
that's a good question --

my intent is for the SDK to clear the JWT internally